Source/WebCore/platform/graphics/cg/GraphicsContextGLCG.cpp

```diff
@@ -349,27 +349,31 @@ bool GraphicsContextGLImageExtractor::extractImage(bool premultiplyAlpha, bool i
     // image data.
     CGColorSpaceRef colorSpace = CGImageGetColorSpace(decodedImage->platformImage().get());
     CGColorSpaceModel model = CGColorSpaceGetModel(colorSpace);
-    if (model == kCGColorSpaceModelIndexed) {
+    bool isColorSpaceModelIndexed = model == kCGColorSpaceModelIndexed;
+    auto destinationCGColorSpace = ignoreGammaAndColorProfile ? sRGBColorSpaceRef() : destinationColorSpace.platformColorSpace();
+    bool needsColorSpaceConversion = !ignoreGammaAndColorProfile && !CFEqual(colorSpace, destinationCGColorSpace);
```

Comment:
> You can check DestinationColorSpaceCG for the comparison.

What is this "DestinationColorSpaceCG"?

> The ignoreGammaAndColorProfile seems to be addressed above, so you may check if you need that.

If you mean at lines 331-332 above, that addresses the case where ignoreGammaAndColorProfile is true when first importing the generic `Image` data into a local `NativeImage` copy.
In this block here, the ignoreGammaAndColorProfile flag is checked (again) to potentially disable the colorspace conversion to a given `unpackColorSpace`, in which case if we just handle the indexed color conversion to sRGB as the code was doing before.

> It may be ok to just compare the WebCore::DestinationColorSpace instances

Oh, I see that I can access the `decodedImage->colorSpace()`, and there's `operator==` that compares the underlying CGColorSpace. Easier indeed, thank you.